src/EFCore.PG/Migrations/NpgsqlMigrationsSqlGenerator.cs

```diff
@@ -194,6 +194,73 @@ protected override void Generate(MigrationOperation operation, IModel model, Mig
                     .Append(")");
             }
 
+            // PostgreSQL-XL - Distribute by (https://www.postgres-xl.org/documentation/sql-createtable.html)
+            if (operation[PostgresXlDistributeByAnnotationNames.DistributeBy] is string)
+            {
+                var distributeBy = new PostgresXlDistributeBy(operation);
+
+                var strategy = distributeBy.DistributionStrategy;
+                var distributeByColumn = distributeBy.DistributeByColumnName;
+                var distributeByFunction = distributeBy.DistributeByColumnFunction;
+                var distributionStyle = distributeBy.DistributionStyle;
+
+                if (distributionStyle == PostgresXlDistributionStyle.None)
+                {
+                    if (strategy == PostgresXlDistributeByStrategy.Replication
+                        || strategy == PostgresXlDistributeByStrategy.Roundrobin
+                        || (distributeByFunction != PostgresXlDistributeByColumnFunction.None
+                            && !string.IsNullOrWhiteSpace(distributeByColumn)))
+                    {
+                        if (strategy == PostgresXlDistributeByStrategy.Replication || strategy == PostgresXlDistributeByStrategy.Roundrobin)
```

Comment:
This seems to repeat the condition above - you can probably just remove the above one, no?